app/scripts/metamask-controller.js

```diff
@@ -5748,6 +5755,154 @@ export default class MetamaskController extends EventEmitter {
     return this.sortEvmAccountsByLastSelected(ethAccounts);
   }
 
+  /**
+   * Handles Hyperliquid referral approval flow.
+   * Shows approval confirmation screen if needed and manages referral URL redirection.
+   * This can be triggered by connection permission grants or existing connections.
+   *
+   * @param {object} req - The request object containing origin, tabId, and optional triggerType.
+   */
+  async handleHyperliquidReferral(req) {
```

Comment:
req is a special param/var name that generally always represents a JSON-RPC request object (the request coming from a dapp). It may be confusing to others seeing `req` represent something else here. Thoughts on naming this something else or just accepting multiple named params rather than a singular param object?